hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieMergeHandle.java

```diff
@@ -357,7 +357,6 @@ public void write(GenericRecord oldRecord) {
         }
       }
 
-      keyToNewRecords.clear();
       ((ExternalSpillableMap) keyToNewRecords).close();
```

Comment:
The `close` method already release the `inMemoryMap` so that we can remove the invocation of `clear`.